processing/src/main/java/io/druid/query/aggregation/AggregatorFactory.java

```diff
@@ -32,13 +32,13 @@
  * provided to the Aggregator through the MetricSelector object, so whatever creates that object gets to choose how
  * the data is actually stored and accessed.
  */
-public interface AggregatorFactory
+public abstract class AggregatorFactory
```

Comment:
yes, that's what I'm saying. (Although I think the JS aggregator would need to override too, not just the histograms)

I don't feel strongly enough to override you. Feel free to keep it the way it is.

Fwiw the reason it is a general preference for me is that I like having an interface that you can implement if you feel that the additional layer of an abstract class is more of a hassle than it's worth. Perhaps because you feel it's simpler for all the methods to be implemented in one file. Or perhaps because you want to be forced to consider how to implement new methods in the subclass when they're added to the interface, instead of accidentally getting the new base implementation that may or may not be appropriate.